### PR TITLE
v1.0

### PR DESCRIPTION
**Pull request by todoubled**
- JSON response parsing
- API methods and tests
- ready to be published to `npm`

### CODE REVIEW

File: test/api.coffee [hunk 1]
```diff
@@ -14,11 +14,13 @@ describe "Intuit API Client", ->
       assert.notEqual intuit.institutions, undefined
 
   describe "getInstitutionDetails", ->
-    it "should should send a signed request", (done) ->
+    it "should should send a signed JSON request", (done) ->
```
Comment (sudothinker) [inline test/api.coffee]: wanna verify the response?

**Comment by todoubled** [comment]
I'm waiting for support help on the remaining calls (3) but this should be good to review and merge for now @hellodigit/engineering

File: test/lib/oauth.coffee [hunk 2]
```diff
@@ -3,27 +3,23 @@ bond = require "bondjs"
...
 
-    it "should return an OAuth token", ->
+    it "should make an Authorized request to return an OAuth token without error", ->
       assert.equal @err, null
       assert @token
       assert @secret
 
-    it "should have an Authorization header with the consumer key", ->
-      regex = new RegExp "#{config.consumerKey}"
-      assert regex.test @spy.calledArgs[0][0].headers.Authorization
```
Comment (todoubled) [inline test/lib/oauth.coffee]: This header gets asserted by the fixture loaded by nock so a successful request means the header matched

**Comment by todoubled** [comment]
@hellodigit/engineering This is ready for review, it's all set for a v1.0 release

**Comment by todoubled** [comment]
@hellodigit/engineering ready for review

File: lib/client.coffee [hunk 3]
```diff
@@ -9,29 +20,74 @@ module.exports = class IntuitClient
     [body, done] = [{}, body] if typeof(body) is "function"
     new Request(@options)[method]("#{BASE_URL}#{path}", body, done)
 
-  institutions: (done) ->
-    @request "get", "/institutions", done
+  institutions: (userId, done) ->
+    @options.userId = userId
```
Comment (sudothinker) [inline lib/client.coffee]: It worth making userId a required attribute on the class initializer? not too sure how you get the userId (the example just sets it). This is the company id or individual user ids?

Comment (todoubled) [inline lib/client.coffee]: Individual userId. Maybe there's a better way to accomplish this, but essentially I want the flexibility of passing in any userId here which is just referenced from the global instance options when signing the SAML request to get the OAuth token. This is the only option that is used in the signature that isn't global

**Comment by sudothinker** [comment]
:+1:

**Pull request merged**